pandas/tests/indexing/test_loc.py

```diff
@@ -43,6 +43,15 @@
 from pandas.tests.indexing.common import Base
 
 
+def test_not_change_nan_loc():
```

Comment:
Could you also add a separate test function for the `df = pd.DataFrame({'A': [np.nan, np.nan, 'b']})` case in the issue as well?

Reply:
Done

Reply:
Could you use `pytest.mark.parameterize` to turn this into 1 test?

Reply:
yes